### PR TITLE
chore: Upgrade antd@4.10.3

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Upgrades Antd to version 4.10.3, in order to use property `maxTagCount="responsive"` in the Select component enabling dropdowns to display tags responsively.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
N.A.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/dashboard/nativeFilters.test.ts [hunk 1]
```diff
@@ -199,7 +199,7 @@ describe('Native filters', () => {
     cy.preserveLogin();
   });
 
-  describe('Nativefilters tests initial state required', () => {
+  describe.only('Nativefilters tests initial state required', () => {
```
Comment (michael-s-molina) [inline superset-frontend/cypress-base/cypress/integration/dashboard/nativeFilters.test.ts:202]:
```suggestion
  describe('Nativefilters tests initial state required', () => {
```

**Comment by michael-s-molina** [comment]
@geido Adding the code below in `superset-ui-core/test/__mocks__/resizer-observer-polyfill.ts` should fix the test error:

```
export default function ResizeObserver(callback: ObserveCallback) {
  return {
    disconnect() {
      allCallbacks.splice(allCallbacks.indexOf(callback), 1);
    },
    observe() {
      if (callback) {
        allCallbacks.push(callback);
      }
    },
    unobserve() { <-- ADD THIS
      if (callback) {
        allCallbacks.splice(allCallbacks.indexOf(callback), 1);
      }
    },
  };
}
```

**Review by michael-s-molina — APPROVED** [review]
Code LGTM.

**Review by codyml — APPROVED** [review]
LGTM - clicked around home, Explore Dashboard view/edit, datasets, SQL Lab and everything looked fine.

**Review by jinghua-qa — APPROVED** [review]
LGTM, run automation and did not see any blocker

**Pull request merged**